### DISCUSSION TITLE
Persisting changes after docker restart

### DISCUSSION THREAD

**Discussion by atomtec** [Cinnamon/kotaemon#343] [Q&A] [answered]
Hi , 
I am running via docker 
docker run \
-e GRADIO_SERVER_NAME=0.0.0.0 \
-e GRADIO_SERVER_PORT=7860 \
-p 7860:7860 -it --rm \
--platform linux/arm64 \
ghcr.io/cinnamon/kotaemon:main-lite

How do i persist the changes after restarting docker , like the files already indexed .

**Comment by mdaca** [comment]
You can use docker volumes to achieve this.  There is an example docker compose here that also enables OIDC:

https://github.com/mdaca/kotaemon-docker-compose-oidc

**Reply by mdaca** [reply]
Here is another example without the OIDC wrapper: 

https://github.com/mdaca/kotaemon-docker-compose

**Comment by taprosoft** [comment, marked as answer]
Hi @atomtec you can mount the `ktem_app_data` to external directory outside of the Docker container to persist its state. Please check this issue for more detail https://github.com/Cinnamon/kotaemon/issues/363

**Comment by drodel** [comment]
As mentioned above by @taprosoft, you can mount the docker container to a directory with the -v or --volume option.
    -v or --volume \<source\>:\<target\>
  
These are the specific steps for your reference.

- make your data directory
mkdir ~/kotaemon-app-data

- start docker pointing the ktem_app_data directory at the local data directoy
docker run  \\
-e GRADIO_SERVER_NAME=0.0.0.0 \\
-e GRADIO_SERVER_PORT=7860 \\
-p 7860:7860 -it --rm \\
--platform linux/arm64 \\
-v ~/kotaemon-app-data:/app/ktem_app_data/ \\
ghcr.io/cinnamon/kotaemon:main-lite
